apps/src/lib/client/tx.rs

```diff
@@ -1338,6 +1389,42 @@ fn convert_amount(
     (asset_type, amount)
 }
 
+/// Freeze a Builder into the format necessary for inclusion in a Tx. This is
+/// the format used by hardware wallets to validate a MASP Transaction.
+struct WalletMap;
+
+impl<P1>
+    masp_primitives::transaction::components::sapling::builder::MapBuilder<
+        P1,
+        ExtendedSpendingKey,
+        (),
+        ExtendedFullViewingKey,
+    > for WalletMap
+{
+    fn map_params(&self, _s: P1) {}
+
+    fn map_key(&self, s: ExtendedSpendingKey) -> ExtendedFullViewingKey {
+        s.to_extended_full_viewing_key()
```

Comment:
This is deprecated with `.to_diversifiable_full_viewing_key`. If we don't need to change it in this PR, let's add `#[allow(deprecated)]` on it and open an issue

Reply:
opened #1334